pkgs/data/fonts/nerd-fonts/update.py

```diff
@@ -0,0 +1,90 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -p python3 -i python3
+import os
+import urllib.request as ureq
+import json
+
+if not all(
+    f"UPDATE_NIX_{v}" in os.environ
+    for v in ["NAME", "PNAME", "OLD_VERSION", "ATTR_PATH"]
+) or not os.environ['UPDATE_NIX_ATTR_PATH'].startswith("nerd-fonts."):
+    raise Exception(
+        "Please don't run this script manually, only with:\n"
+        "nix-shell maintainers/scripts/update.nix --argstr path nerd-fonts "
+        "--argstr commit true"
+    )
+
+RELEASE_INFO_URL = "https://api.github.com/repos/ryanoasis/nerd-fonts/releases/latest"
+FONTS_INFO_URL_TEMPLATE = "https://raw.githubusercontent.com/ryanoasis/nerd-fonts/refs/tags/{}/bin/scripts/lib/fonts.json"
+SHA256_URL_TEMPLATE = "https://github.com/ryanoasis/nerd-fonts/releases/download/{}/SHA-256.txt"
+
+RELEASE_INFO_FILENAME = "release.json"
+FONTS_INFO_FILENAME = "fonts.json"
+CHECKSUMS_FILENAME = "checksums.json"
+
+def fetchjson(url):
+    with ureq.urlopen(url) as r:
+        return json.loads(r.read())
```

Comment:
Nit: This and the other `open` trigger https://docs.astral.sh/ruff/rules/unspecified-encoding/. Suggest to add `, encoding="utf-8"` to both.